### PR TITLE
Update dependencies with pip-compile

### PR DESCRIPTION
**Pull request by madig**
Closes https://github.com/googlefonts/googlefonts-project-template/issues/75.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -48,5 +48,7 @@ clean:
 update-project-template:
 	npx update-template https://github.com/googlefonts/googlefonts-project-template/
 
-update:
-	pip install --upgrade $(dependency); pip freeze > requirements.txt
+update: venv
+	venv/bin/pip install --upgrade pip-tools
+	venv/bin/pip-compile --upgrade --resolver=backtracking requirements.in
```
Comment (anthrotype) [inline Makefile:53]: what does --resolver=backtracking do?

File: requirements.txt [hunk 2]
```diff
@@ -1,6 +1,2 @@
-fontmake>=2.4
-fontbakery>=0.8.11
-gftools[qa]>=0.9.23
-drawbot-skia>=0.4.8
-sh>=1.14.2
-bumpfontversion>=0.2.0
+# Placeholder file, update the requirements by running `make update`.
+-r requirements.in
```
Comment (anthrotype) [inline requirements.txt:2]: is this "requirements.txt" file even needed in the template repo, now that it's going to be auto-regenerated?

File: Makefile (hunk 1 above)
Comment (madig) [reply Makefile:53]: Not sure, but pip-compile prints a warning that it's becoming the default in the next version and that you should pass the flag.

File: requirements.txt (hunk 2 above)
Comment (madig) [reply requirements.txt:2]: I guess it's needed for when you `make build` without updating first.

**Comment by madig** [comment]
Oh, I just saw that first-run.py also pins deps. Need to deal with that.

File: requirements.txt (hunk 2 above)
Comment (anthrotype) [reply requirements.txt:2]: LMGTFY https://pip-tools.readthedocs.io/en/latest/#a-note-on-resolvers

File: requirements.in [hunk 3]
```diff
@@ -0,0 +1,6 @@
+fontmake>=2.4
+fontbakery>=0.8.11
```
Comment (felipesanches) [inline requirements.in:2]: `fontbakery[googlefonts]>=0.9.2`

**Comment by simoncozens** [comment]
taken over in #169.

**Pull request closed without merging**